rickchurch/church.py

```diff
@@ -105,22 +110,37 @@ async def authorize() -> fastapi.Response:
 @app.get("/oauth_callback", include_in_schema=False)
 async def auth_callback(request: fastapi.Request) -> fastapi.Response:
     """This endpoint is only used as a redirect target from discord OAuth2."""
+    client = request.app.state.client
     code = request.query_params["code"]
     try:
-        user = await get_oauth_user(code)
+        user, access_token = await get_oauth_user(client, code)
         token = await add_user(user, request.state.db_conn)
     except PermissionError:
         # `add_user` can return `PermissionError` if the user already has a token, which is banned.
         raise fastapi.HTTPException(401, "You are banned")
 
+    # Join them into the rickchurch server
+    res = await client.put(
+        constants.DISCORD_BASE_URL + f"/guilds/{constants.discord_guild_id}" + f"/members/{user['id']}",
```

Comment:
```suggestion
        f"{constants.DISCORD_BASE_URL}/guilds/{constants.discord_guild_id}/members/{user['id']}",
```
There is no reason to split this into 3 f-strings and concat them together, if they can just be in a single f-string.